plugins/outputs/nats/nats.go

```diff
@@ -85,7 +100,47 @@ func (n *NATS) Connect() error {
 
 	// try and connect
 	n.conn, err = nats.Connect(strings.Join(n.Servers, ","), opts...)
+	if n.Jetstream != nil {
+		// connect to jetstream
+		n.jetstreamClient, err = jetstream.New(n.conn)
+		if err != nil {
+			return fmt.Errorf("failed to connect to jetstream: %w", err)
+		}
+		if n.Jetstream.Stream == "" {
+			return errors.New("stream cannot be empty")
+		}
+		streamExists := n.streamExists(n.Jetstream.Stream)
+		if !streamExists {
+			if !n.Jetstream.AutoCreateStream {
+				return fmt.Errorf("stream %s does not exist", n.Jetstream.Stream)
+			}
+			streamConfigJSON := strings.TrimSpace(n.Jetstream.StreamJSON)
+			var streamCfg jetstream.StreamConfig
+			if len(streamConfigJSON) > 0 {
+				err = json.Unmarshal([]byte(streamConfigJSON), &streamCfg)
+				if err != nil {
+					return fmt.Errorf("invalid jetstream config %w", err)
+				}
+			}
+			streamCfg.Name = n.Jetstream.Stream
+			streamCfg.Subjects = []string{n.Subject}
+			n.Jetstream.streamConfig = streamCfg
+			err = n.createStream(streamCfg)
+			if err != nil {
+				return fmt.Errorf("failed to create stream: %w", err)
+			}
+		}
+	}
+	return err
+}
+
+func (n *NATS) streamExists(stream string) bool {
+	_, err := n.jetstreamClient.Stream(context.Background(), stream)
+	return err == nil
+}
 
+func (n *NATS) createStream(streamCfg jetstream.StreamConfig) error {
+	_, err := n.jetstreamClient.CreateStream(context.Background(), streamCfg)
```

Comment:
Do you use these above helper functions in testing? Otherwise I would say skip them since they are not doing anything more than one function, and only called once.